src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala

```diff
@@ -280,8 +280,53 @@ object MutableSorting extends Sorting[Array, Int] {
 
   }
 
-  def radixSort(xs: Array[Int]): Array[Int] = ???
+  /**
+    * @author Xoeseko
+    * Radix Sort is a sorting algorithm that sorts an array of numbers into according to each number's most significant
+    * symbols (bits, digits...) depending on the given base
+    * @param xs an array to be sorted
+    * @param base the sorting base and also the number of buckets
+    * @return the sorted array
+    */
+    def radixSort(xs: Array[Int], base: Int = 10): Array[Int] = {
+      var array = xs
+      def listToBuckets(arr: Array[Int], b: Int, it: Int): Array[Array[Int]] = {
+        val buckets = Array.ofDim[ArrayBuffer[Int]](base)
+        // Initialize buckets
+        for (i <- 0 until buckets.length){
+          buckets(i) = new ArrayBuffer[Int]()
+        }
+        for (n <- arr){
+          // Start by getting the current digit and assigning the number to corresponding bucket.
+          val digit = (n / math.pow(b, it).toInt) % b
+          buckets(digit) += n
+        }
+      buckets.map(a => a.toArray)
+      }
 
+      def bucketsToList(buckets: Array[Array[Int]]): Array[Int] ={
```

Comment:
Should it be declared `private`? Would it be better to nest this method inside the `radixSort()` function?

Reply:
It is already embedded within the radixsort function isn't it ? Do I need to also make it private ? Why then if it is already encapsulated..?

Reply:
Sorry, my bad. ;)